src/PrestaShopBundle/Twig/Layout/LayoutBuilder.php

```diff
@@ -0,0 +1,188 @@
+     * TODO: Need to be removed when we will render kpis with twig only!
+     */
+    public function enableSmartyNewTheme(): void
+    {
+        $this->context->getContext()->smarty->setTemplateDir(_PS_BO_ALL_THEMES_DIR_ . 'new-theme/template/');
+    }
+
+    public function getIsoUser(): string
+    {
+        return $this->context->getLanguage()->getIsoCode();
+    }
+
+    public function isRtlLanguage(): bool
+    {
+        return (bool) $this->context->getLanguage()->isRTL();
+    }
+
+    public function getControllerName(): string
+    {
+        return htmlentities(Tools::getValue('controller'));
+    }
+
+    public function isMultiShop(): bool
+    {
+        return Shop::isFeatureActive();
+    }
+
+    public function isCollapseMenu(): bool
+    {
+        $cookie = $this->context->getContext()->cookie;
+
+        if (isset($cookie->collapse_menu)) {
+            return boolval($cookie->collapse_menu);
+        }
+
+        return false;
+    }
+
+    public function getJsRouterMetadata(): array
+    {
+        return [
+            // base url for javascript router
+            'base_url' => $this->requestStack->getCurrentRequest()->getBaseUrl(),
+            //security token for javascript router
+            'token' => $this->tokenManager->getToken($this->userProvider->getUsername())->getValue(),
+        ];
+    }
+
+    public function isDebugMode(): bool
+    {
+        return $this->debugMode;
+    }
+
+    public function isInstallDirExists(): bool
+    {
+        return file_exists(_PS_ADMIN_DIR_ . '/../install');
+    }
+
+    public function getVersion(): string
+    {
+        return $this->psVersion;
+    }
+
+    public function getDefaultTabLink(): string
+    {
+        /** @var Tab $tab */
+        $tab = $this->tabRepository->findOneBy(['id' => (int) $this->context->getContext()->employee->default_tab]);
+
+        return $this->context->getLegacyAdminLink($tab->getClassName());
+    }
+
+    public function isMaintenanceMode(): bool
+    {
+        return !(bool) $this->configuration->get('PS_SHOP_ENABLE');
+    }
+
+    public function isMaintenanceAllowAdmins(): bool
+    {
+        return (bool) $this->configuration->get('PS_MAINTENANCE_ALLOW_ADMINS');
+    }
+
+    public function getDisplayBackOfficeTop(): ?string
+    {
+        if ($this->displayBackOfficeTop) {
+            return $this->displayBackOfficeTop;
+        }
+
+        $renderedHook = $this->hookDispatcher->dispatchRenderingWithParameters('displayBackOfficeTop');
+
+        if (!$content = $renderedHook->getContent()) {
+            return null;
+        }
+
+        foreach ($content as $hookContent) {
+            if (is_array($hookContent)) {
+                $this->displayBackOfficeTop .= implode($hookContent);
+            } else {
+                $this->displayBackOfficeTop = $hookContent;
+            }
+        }
+
+        return $this->displayBackOfficeTop;
+    }
+
+    public function getCurrentTabLevel(): int
```

Comment:
As suggested in another comment, this should be transformed into a `displayWithTabs` getter